### DISCUSSION TITLE
Templating specific keys in arbitrarily nested JSON

### DISCUSSION THREAD

**Discussion by jpluscplusm** [hairyhenderson/gomplate#2052] [Q&A]
Hey all 👋 

So, I'm being handed some JSON in a pretty nested form, and I'd like to:

1. template the value of the `Chapter.content` field, wherever it appears in the nested data;
2. emit values which recognise that they're inside a JSON string, hence must be escaped correctly.

I guess that, if there were a way of satisfying point (2) then I could simply template the stringified form of the JSON document in one go, but that /feels/ tricky to do.

Here's an example of the form. The `sub_items` list could be nested more deeply than this example, but I won't know its structure in advance:

```JSON
[
    {
        "someKey": "some value"
    },
    {
        "sections": [
            "Separator",
            {
                "PartTitle": "foo"
            },
            {
                "Chapter": {
                    "name": "Chapter 1",
                    "content": "This might be a template",
                    "sub_items": [
                        "Separator",
                        {
                            "PartTitle": "bar"
                        },
                        {
                            "Chapter": {
                                "name": "Chapter 1.1",
                                "content": "This also might be a template",
                                "sub_items": []
                            }
                        }
                    ]
                }
            }
        ]
    }
]
```

Is this achievable using Gomplate?

If not, what's the smallest structural modification I could make to the data such that it *would* be possible?

### Small addendum

Is it possible to **only** emit the (templated, as discussed above) 2nd list item from the top-level list I'm given?

**Comment by hairyhenderson** [comment]
If I'm understanding this correctly you just want to have a JSON file where some strings might contain template strings?

If so, you just need to treat the JSON file as a template file itself and run it through gomplate as usual. Something like:

_input.json:_
```json
[
    {
        "sections": [
...
                "Chapter": {
                    "content": "{{ printf "some probably more complex template string" }}",
                    "sub_items": [
                        {
                            "Chapter": {
                                "content": "{{ ... }}",
                            }
...
]
```

If you wanted to reference data defined in the JSON file itself you could use the file as both the template _and_ a datasource or context value (though you'd be referencing the pre-rendered template):

```console
$ gomplate -c data=input.json -f input.json
...
```

> Is it possible to **only** emit the (templated, as discussed above) 2nd list item from the top-level list I'm given?

I don't know that I understand what you mean here, but check out the `range` operator and/or `index`.
